app/models/elastic_search.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+class ElasticSearch
+  def initialize(_index = nil)
+    @es_config = MlnConfigurationController.new.elasticsearch_config('teachersets')
+    arguments = {
+      host: @es_config['host'],
+      transport_options: {
+        request: { open_timeout: @es_config['connect_timeout'] },
+        headers: { content_type: 'application/json' }
+      }
+    }
+    @client = Elasticsearch::Client.new(arguments)
+    @current_file = File.basename(__FILE__)
+    @index = @es_config['index'] || 'teacherset'
+    @type = @es_config['type'] || 'teacherset'
+  end
+
+
+  def index_doc_count
+    response = @client.perform_request 'GET', @index + '/_count'
+    total_docs = response.body['count']
+    total_docs
+  end
+
+
+  def create_document(id, body)
+    @client.create index: @index, type: @type, id: id, body: body
+  end
+
+
+  def delete_document_by_id(id)
+    response = @client.delete index: @index, type: @type, id: id
+    LogWrapper.log('DEBUG', {'message' => "ES document successfully deleted. Id: #{id}", 
+                             'method' => 'delete_document_by_id'})
+
+    response
+  end
+
+
+  def partial_search_by_query(keyword, from, size)
+    query = {"from": from, "size": size, "query": {"multi_match": {"query": keyword, "fields": %w[title description contents]}}}
+    search_by_query(query)
+  end
+
+
+  def fuzzy_search_by_query(keyword, from, size, fuzzy_val)
+    query = {"from": from, "size": size, "query": {"multi_match": 
```

Comment:
In addition to method documentation, I feel this line is important.  It sets up the rules by which we're searching.  Would you mind adding some documentation to explain what we're doing and your thought process?